### PR TITLE
feat(smn): add new data_source message_templates

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/smn' TESTARGS='-run TestAccDatasourceSmnMessageTemplates_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/smn -v -run TestAccDatasourceSmnMessageTemplates_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceSmnMessageTemplates_basic
=== PAUSE TestAccDatasourceSmnMessageTemplates_basic
=== CONT  TestAccDatasourceSmnMessageTemplates_basic
--- PASS: TestAccDatasourceSmnMessageTemplates_basic (23.98s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/smn       24.018s
```

### CODE REVIEW

File: docs/data-sources/smn_message_templates.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Use this data source to get the list of SMN message templates.
+
+## Example Usage
+
+```hcl
+variable "name" {}
+
+data "huaweicloud_smn_message_templates" "test"{
+  name     = var.name
```
Comment (houpeng80) [inline docs/data-sources/smn_message_templates.md]:
```suggestion
  name = var.name
```

File: docs/data-sources/smn_message_templates.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `templates` - The list of message templates.
+  The [MessageTemplate](#SmnMessageTemplate_MessageTemplate) structure is documented below.
+
+<a name="SmnMessageTemplate_MessageTemplate"></a>
+The `MessageTemplate` block supports:
```
Comment (houpeng80) [inline docs/data-sources/smn_message_templates.md]:
```suggestion
* `templates` - The list of message templates.
  The [templates](#SmnMessageTemplate_MessageTemplate) structure is documented below.

<a name="SmnMessageTemplate_MessageTemplate"></a>
The `templates` block supports:

```

File: docs/data-sources/smn_message_templates.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `name` - (Optional, String) Specifies the name of the message template.
+
+* `protocol` - (Optional, String) Specifies the protol of the message template.
```
Comment (houpeng80) [inline docs/data-sources/smn_message_templates.md]:
```suggestion
* `protocol` - (Optional, String) Specifies the protocol of the message template.

```

File: huaweicloud/services/smn/data_source_huaweicloud_smn_message_templates.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the name of the message template.`,
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the protol of the message template.`,
```
Comment (houpeng80) [inline huaweicloud/services/smn/data_source_huaweicloud_smn_message_templates.go]: ditto

File: docs/data-sources/smn_message_templates.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/smn_message_templates.md]: fixed

File: docs/data-sources/smn_message_templates.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/smn_message_templates.md]: fixed

File: docs/data-sources/smn_message_templates.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/smn_message_templates.md]: fixed

File: huaweicloud/services/smn/data_source_huaweicloud_smn_message_templates.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/smn/data_source_huaweicloud_smn_message_templates.go]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/smn' TESTARGS='-run TestAccDatasourceSmnMessageTemplates_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/smn -v -run TestAccDatasourceSmnMessageTemplates_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceSmnMessageTemplates_basic
=== PAUSE TestAccDatasourceSmnMessageTemplates_basic
=== CONT  TestAccDatasourceSmnMessageTemplates_basic
--- PASS: TestAccDatasourceSmnMessageTemplates_basic (24.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/smn       24.932s
```

File: docs/data-sources/smn_message_templates.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+* `protocol` - (Optional, String) Specifies the protocol of the message template.
+
+* `template_id` - (Optional, String) Specifies the message template ID.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
```
Comment (ShiChangkuo) [inline docs/data-sources/smn_message_templates.md]:
```suggestion
* `id` - The data source ID.

```

Comment (Jason-Zhang9309) [reply docs/data-sources/smn_message_templates.md]: fixed

**Pull request merged**